api4/user.go

```diff
@@ -109,6 +110,38 @@ func updateUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func updatePassword(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	props := model.MapFromJson(r.Body)
+	currentPassword := props["current_password"]
+	if len(currentPassword) <= 0 {
+		c.SetInvalidParam("current_password")
+		return
+	}
+
+	newPassword := props["new_password"]
+
+	c.LogAudit("attempted")
+
+	if c.Params.UserId != c.Session.UserId {
```

Comment:
Why does the endpoint require a userid then? Shouldn't we just get it from the session?